dipy/reconst/multi_voxel.py

```diff
@@ -35,9 +36,29 @@ def multi_voxel_fit(single_voxel_fit):
 
     def new_fit(self, data, mask=None, **kwargs):
         """Fit method for every voxel in data"""
-        # If only one voxel just return a normal fit
+        # Analyze if the single voxel fit has any "non standard"
+        # key-word arguments:
+        arg_spec = getargspec(single_voxel_fit)
+        # List of the arguments to the function:
+        this_args = arg_spec.args
+        # The following are "standard" arguments:
+        this_args.pop(this_args.index("self"))
+        this_args.pop(this_args.index("data"))
+        if "mask" in this_args:
+            this_args.pop(this_args.index("mask"))
+        # What remains are arguments that are defined separate from **kwargs
+        # (kwargs are specifically reserved only for parallelization
+        # arguments):
+        func_kwargs = {}
+        for kwarg in this_args:
+            if kwarg in kwargs:
+                func_kwargs.update({kwarg: kwargs[kwarg]})
+                kwargs.pop(kwarg)
```

Comment:
We are switching the code bad to keyword-only arguments ( [PEP 3102 ](https://peps.python.org/pep-3102/)).

Many PRs has been merged and the reconst module will be merge soon. https://github.com/dipy/dipy/pull/3257

I believe this code will not work with keyword only arguments since they are not in `arg_spec.args`.

I need to test it, but can you handle this case (which will become the general case)?